added login tests

andriy-grytsiv opened the pull request:
1. added 7 login tests
2. added relevant date providers
3. ont test is not implemented as I can NOT catch the tooltip

I feel like there is way too much copy/paste in code. so please let me know how to improve this

andriy-grytsiv committed:
Added login tests

andriy-grytsiv committed:
Refactored login tests using Page Objects approach

andriy-grytsiv committed:
1. changed implementation to static Page Objects
2. added PageBase, improved TestBase
3. implemented multiple browsers support
4. moved locators to property files

andriy-grytsiv committed:
1. added catalog tests
2. added shopping cart tests

Fulcrum81 commented on src/test/java/tests/CatalogTest.java:
```diff
@@ -0,0 +1,91 @@
+public class CatalogTest extends TestBase{
+    @Test
+    public static void navigateToCatalogCategory() {
+        String expectedCategoryTitle = "Rubber Ducks | My Store";
```
ожидаемые тайтлы в принципе можно на страничных классах прописать, и оттуда буля возвращать в тесты

andriy-grytsiv replied:
понял. я ориентировался на то, что тестовые данные лучше держать в самом тесте. на уровне PageBase реализован метод, в который надо передать expectedPageTitle, он сравнивает с фактическим и возращает буль.  какой подход используется на реальных проектах? или это дело вкуса?

Fulcrum81 commented on src/test/java/tests/LoginTest.java:
```diff
@@ -0,0 +1,155 @@
+        String emptyLoginFieldsMessage = "Please fill in this field";
+
+        // start test
+        getDriver();
```
а почему он тут? как же beforeMethod?

andriy-grytsiv replied:
скорее всего это остатки перехода с обьектного на статический PageObject, пропустил в части тестов. перепровел, убрал лишние вызовы getDriver(). оставил только те, где getDriver() используется для открытия страницы с кастомной линкой. прогнал тесты, все работает

andriy-grytsiv committed:
1. added allure reporting support with screenshots, environment properties and other stuff
2. added report portal support which is NOT working and I can't find the reason
3. added testNG xml configuration and added them to build steps
